ios/brave-ios/Sources/PlaylistUI/PlayerView.swift

```diff
@@ -0,0 +1,353 @@
+            Task { await model.seekForwards() }
+          } label: {
+            Label("Step Forward", braveSystemImage: "leo.forward.15")
+          }
+          .buttonStyle(.playbackControl(size: .large))
+        }
+        .ignoresSafeArea(edges: .vertical)
+      }
+      .buttonStyle(.playbackControl)
+      .tint(Color(braveSystemName: .textPrimary))
+      .backgroundStyle(Color.white.opacity(0.2))
+      .task(priority: .low) {
+        self.currentTime = model.currentTime
+        for await currentTime in model.currentTimeStream {
+          self.currentTime = currentTime
+        }
+      }
+    }
+  }
+}
+
+@available(iOS 16.0, *)
+struct VideoAmbianceBackground: View {
+  var playerModel: PlayerModel
+
+  @State private var videoAmbianceDecorationImage: UIImage?
+
+  var body: some View {
+    VStack {
+      if let videoAmbianceDecorationImage {
+        Image(uiImage: videoAmbianceDecorationImage)
+          .resizable()
+          .blur(radius: 30)
+          .id(videoAmbianceDecorationImage)
+          .transition(.opacity)
+      }
+    }
+    .task(priority: .medium) {
+      for await image in playerModel.videoAmbianceImageStream {
+        withAnimation {
+          videoAmbianceDecorationImage = image
+        }
+      }
+    }
+  }
+}
+
+@available(iOS 16.0, *)
+struct CompactMediaScrubberLabel: View {
+  var currentTime: TimeInterval
+  var duration: TimeInterval
+
+  private var currentValueLabel: Text {
+    return Text(.seconds(currentTime), format: .time(pattern: .minuteSecond))
+  }
+
+  private var remainingTimeLabel: Text {
+    return Text(.seconds(currentTime - duration), format: .time(pattern: .minuteSecond))
+  }
+
+  var body: some View {
+    HStack {
+      currentValueLabel
+      Text(verbatim: "/")  // FIXME: Does this need some sort of localization?
+      remainingTimeLabel
+    }
+    .font(.caption2)
+  }
+}
+
+/// A custom Layout which accepts an optional aspect ratio to use when sizing the VideoPlayer.
+///
+/// If an aspect ratio is supplied then the underlying VideoPlayer will attempt to size itself to
+/// fill the width of the parent while maintaining that aspect ratio. In the case where there is not
+/// enough vertical space to maintain the supplied aspect ratio, the height will be lowered and
+/// the aspect ratio may not be respected and black bars may show up on the video.
+///
+/// If no aspect ratio is supplied then the underlying VideoPlayer will simply fill the entire
+/// available space like it would if no layout was used.
+///
+/// This is neccessary because there is no way to conditionally apply an `aspectRatio` modifier to
+/// a `View`
+@available(iOS 16.0, *)
+private struct VideoPlayerLayout: Layout {
+  var aspectRatio: CGFloat?
+
+  func sizeThatFits(
+    proposal: ProposedViewSize,
+    subviews: Subviews,
+    cache: inout ()
+  ) -> CGSize {
+    assert(subviews.count == 1)
+    if let aspectRatio {
+      let proposedSize = proposal.replacingUnspecifiedDimensions()
```

Comment:
I am confused here why do we want to change dimensions here CGSize(width: 10, height: 10)

Reply:
Do you mean why do we want to change the layout of the VideoPlayer? The doc on the top of the layout explains the reasoning behind it (`aspectRatio` modifier can't be conditional), if you mean why do we replace the unspecified dimensions, that just lets SwiftUI fill the values with numeric values since `ProposedViewSize`'s width and height are optional values. 10 is simply the default when those values are `nil` which won't happen in this case anyways

Reply:
Yeah I got why we want to change layout. I was confused about dimensions but got it now.